README.md

```diff
@@ -1,231 +1,50 @@
-  the same machine.  This PID is used to get information about limits set for process, etc.
-* `-P` - top directory of COSS, DDAC or DSE installation (for tarball installs)
-* `-r` - remove collected files after generation of resulting tarball
-* `-s` - options to pass to SSH/SCP
-* `-S` - AWS secret for the S3 upload (Can be stored as the `DS_AWS_SECRET` environment variable instead)
-* `-t` (required) - specifies the type of installation: `dse`, `ddac`, `coss`;
-* `-T` - ticket/ref id for the S3 upload and encrypted tarball naming (required for S3 uploads)
-* `-u` - specifies timeout for SSH in seconds (default: 600). You may need to increase it
-  if you're using `extended` collection mode
-* `-v` - enables more verbose output by all scripts
-* `-z` - don't execute commands that require `sudo`
-
-Please note that user name should be passed as `-o User=...` in `-s` option, as `scp` and
-`ssh` are using different ways to pass user name.
-
-Example of usage with list of hosts passed explicitly (file `mhosts`):
-
-```sh
-./collect_diag.sh -t dse -f mhosts -r -s \
-  "-i ~/.ssh/private_key -o StrictHostKeyChecking=no -o User=automaton"
-```
-
-or for DDAC:
 
-```sh
-./collect_diag.sh -t ddac -f mhosts -r -s \
-  "-i ~/.ssh/private_key -o StrictHostKeyChecking=no -o User=automaton" \
-  -P /usr/local/lib/cassandra
+```bash
+# The ISSUE variable is typically a JIRA ID, but can be any unique label
+export ISSUE=<JIRA_ID>
+make
 ```
 
-if it's running on the machine with DSE/DDAC/C*, then it could be as simple as, as it will
-use `nodetool status` to obtain a list of nodes:
+This will generate a _.tar.gz_ tarball with the `issueId` set in the packaged configuration file. The archive will named in the format `ds-collector.$ISSUE.tar.gz`.
 
-```sh
-./collect_diag.sh -t ddac -r -P /usr/local/lib/cassandra
-```
+# Building the Collector with automatic upload ability
 
-In order to encrypt the generated tarball and upload it to S3, specify the necessary arguments:
+If the collector is built with the following variables defined, all collected diagnostic snapshots will be encrypted and uploaded to a specific AWS S3 bucket. Encryption will use a one-off built encryption key, that will be stored in an AWS Secrets Manager.
 
-```sh
-./collect_diag.sh -t coss -K <aws key> -S <aws secret> -T <ticket id> -e /path/to/enc_secret.key -B <s3 bucket name> -r
+```bash
+# AWS Key and secret for S3 bucket, where the diagnostic snapshots will be uploaded to
+export COLLECTOR_S3_AWS_KEY=xxx
+export COLLECTOR_S3_AWS_SECRET=yyy
+# AWS Key and secret for Secrets Manager, where the one-off build-specific encryption key will be stored
+export COLLECTOR_SECRETSMANAGER_KEY=zzz
+export COLLECTOR_SECRETSMANAGER_SECRET=qqq
 ```
 
+After the keys and secrets are defined in the environment, the process to build the collect is the same as described above. To use this feature you will need the aws-cli and openssl installed on your local machine as well.
```

Comment:
"the process to build the collect is" -> "the process to build the collector is"